nowcasting_dataset/dataset/split/split.py

```diff
@@ -50,31 +46,43 @@ def split_data(
 
     logger.info(f"Splitting data with method {method}")
 
-    if method == SplitMethod.Same:
+    datetimes = pd.DatetimeIndex(datetimes)
+
+    if method == SplitMethod.SAME:
         train_datetimes = datetimes
         validation_datetimes = datetimes
         test_datetimes = datetimes
-    elif method == SplitMethod.Day:
-        train_datetimes, validation_datetimes, test_datetimes = split_day(
-            datetimes=datetimes, train_test_validation_split=train_test_validation_split
+    elif method == SplitMethod.DAY:
+        train_datetimes, validation_datetimes, test_datetimes = split_method(
+            datetimes=datetimes,
+            train_test_validation_split=train_test_validation_split,
+            method="modulo",
         )
-    elif method == SplitMethod.DayRandom:
-        train_datetimes, validation_datetimes, test_datetimes = split_day_random(
-            datetimes=datetimes, train_test_validation_split=train_test_validation_split
+    elif method == SplitMethod.DAY_RANDOM:
+        train_datetimes, validation_datetimes, test_datetimes = split_method(
+            datetimes=datetimes,
+            train_test_validation_split=train_test_validation_split,
+            method="random",
         )
-    elif method == SplitMethod.Week:
-        train_datetimes, validation_datetimes, test_datetimes = split_week(
-            datetimes=datetimes, train_test_validation_split=train_test_validation_split
+    elif method == SplitMethod.WEEK:
+        train_datetimes, validation_datetimes, test_datetimes = split_method(
+            datetimes=datetimes,
+            train_test_validation_split=train_test_validation_split,
+            method="modulo",
+            freq="W",
         )
-    elif method == SplitMethod.WeekRandom:
-        train_datetimes, validation_datetimes, test_datetimes = split_week_random(
-            datetimes=datetimes, train_test_validation_split=train_test_validation_split
+    elif method == SplitMethod.WEEK_RANDOM:
+        train_datetimes, validation_datetimes, test_datetimes = split_method(
+            datetimes=datetimes,
+            train_test_validation_split=train_test_validation_split,
+            method="random",
+            freq="W",
         )
-    elif method == SplitMethod.Year:
+    elif method == SplitMethod.YEAR:
         train_datetimes, validation_datetimes, test_datetimes = split_year(
```

Comment:
Should this line, and the following few lines, be changed to something like:

```suggestion
        train_datetimes, validation_datetimes, test_datetimes = split_method(
            datetimes=datetimes,
            train_test_validation_split=train_test_validation_split,
            method="modulo",
            freq="Y",
        )
```

Reply:
yea, althought the year one actually takes the specific years in and splits it like that. Note like the the other methods. So there is something a bit extra in year. 

I could move this year method over to split_method

Reply:
actually I will do that, as the year still has a long loop in it